meson.build

```diff
@@ -1394,36 +1394,38 @@ if features['rpi-mmal']
                      'video/out/vo_rpi.c')
 endif
 
-vaapi = dependency('libva', version: '>= 1.1.0', required: get_option('vaapi'))
-features += {'vaapi': vaapi.found() and features['libdl'] and (features['x11'] or
-             features['wayland'] or egl_drm.allowed())}
+libva = dependency('libva', version: '>= 1.1.0', required: get_option('vaapi'))
 
-if features['vaapi']
-    dependencies += vaapi
-    sources += files('video/filter/vf_vavpp.c',
-                     'video/vaapi.c')
-endif
-
-vaapi_drm = dependency('libva-drm', version: '>= 1.1.0', required: get_option('vaapi-drm').require(features['vaapi']))
-features += {'vaapi-drm': features['vaapi'] and vaapi_drm.found()}
+vaapi_drm = dependency('libva-drm', version: '>= 1.1.0', required: get_option('vaapi-drm').require(libva.found()))
+features += {'vaapi-drm': features['drm'] and vaapi_drm.found()}
```

Comment:
I would also move `features['drm'] ` into `require()` above. But then you need to check twice, because `required: false` would not make `vaapi_drm` not found.

So basing it on the option would probably work, similar to previous
```
libva_drm = dependency('libva-drm', version: '>= 1.1.0', required: get_option('vaapi-drm'))
vaapi_drm = get_option('vaapi-drm').require(libva.found() and features['drm'] and libva_drm.found())
features += {'vaapi-drm': vaapi_drm}
```

And this made me think, we could even optimize configure to not search for libraries that would not be used. For example

```
vaapi_drm = get_option('vaapi-drm').require(libva.found() and features['drm'])
if vaapi_drm
    libva_drm = dependency('libva-drm', version: '>= 1.1.0')
    features += {'vaapi-drm': libva_drm.found()}
    if features['vaapi-drm'] # this conditional may not be required, I think by default meson doesn't return disabler()
        dependencies += libva_drm
    endif
else
       features += {'vaapi-drm': false}
endif
```

but I really dislike this nested conditional and the need to have false branch.

I'm also not meson master, @Dudemanguy may have better ideas. Looking at our `feature` array, I have an idea how to make it easier to handle. Might send RFC later.

Reply:
`required: get_option('vaapi-drm').require(libva.found() and features['drm'])` skips trying to find libva-drm if you disable drm with `-Ddrm=disabled` so I think that change solves the issue or did you mean something else?

Reply:
I don't think so
```
dependency('libva-drm', version: '>= 1.1.0', required: false)
```
will still try to find the library, it will not fail if not found, but still does a lookup. It is not a problem, because configuration is fast anyway, just an optimization.

Reply:
Ah, my bad. I see now that `required` also takes a `feature`, not only `bool`, so `disabled` state is propagated and indeed search for lib would be skipped.

Than it is indeed, enough to put `features['drm']` into `require`.